### PR TITLE
Hemoglobin Exposure Model 3 Artifact

### PR DESCRIPTION
**Pull request by albrja**
### Description
Update to artifact to include hemoglobin exposure with location weights for pregnant and lactating women.  Also includes small changes to loader to clean artifact data that was being subset.
- *Category*: Data artifact
- *JIRA issue*: [MIC-2954](https://jira.ihme.washington.edu/browse/MIC-2954)
- *Research reference*: https://vivarium-research.readthedocs.io/en/latest/concept_models/vivarium_iv_iron/maternal_model/concept_model.html#hemoglobin-distribution-weighting-strategy

Added hemoglobin exposure data to artifact with location weights.  Update to artifact keys that were being subset.

### Verification and Testing
Ran make_artifacts for both South Asia and Sub-Saharan Africa locations.  Loaded artifact data and data looks as expected.  Hemoglobin exposure and location weights sum to 1.

### CODE REVIEW

File: src/vivarium_gates_iv_iron/data/loader.py [hunk 1]
```diff
@@ -170,10 +172,10 @@ def load_asfr(key: str, location: str):
 
     # pivot
     asfr = asfr.reset_index()
-    asfr = asfr[(asfr.sex == 'Female')
-                & (asfr.age_start >= 7)
-                & (asfr.age_end <= 57)
-                & (asfr.year_start == 2019)]
+    # asfr = asfr[(asfr.sex == 'Female')
```
Comment (mattkappel) [inline src/vivarium_gates_iv_iron/data/loader.py]: I don't think that we should change this subsetting (yet).

**Review by mattkappel — APPROVED** [review]

File: src/vivarium_gates_iv_iron/components/hemoglobin.py [hunk 2]
```diff
@@ -16,6 +21,70 @@ def __init__(self):
...
+        self.columns_created = ["country", "hemoglobin_distribution_propensity", "hemoglobin_percentile"]
+        # load data
+        mean = builder.data.load(data_keys.HEMOGLOBIN.MEAN)
+        stddev = builder.data.load(data_keys.HEMOGLOBIN.STANDARD_DEVIATION)
+        # TODO replace --->>>
+        location_weights = mean.copy().drop(columns=["value"])
+        # lookup data, categorical column "country"
+        mean_dfs, stddev_dfs = [], []
+        # TODO have this in artifact and correct the location weights
+        for country in ["Bangladesh", "Pakistan", "India"]:
```
Comment (rmudambi) [inline src/vivarium_gates_iv_iron/components/hemoglobin.py:35]: Store this list of countries somewhere sensible as a constant?

File: src/vivarium_gates_iv_iron/data/loader.py [hunk 3]
```diff
@@ -456,16 +459,66 @@ def load_maternal_disorders_ylds(key: str, location: str) -> pd.DataFrame:
...
+        hemoglobin_data = pd.concat([hemoglobin_data], keys=[loc], names=['location'])
+        country_dfs.append(hemoglobin_data)
+
+    national_level_hemoglobin_data = pd.concat(country_dfs)
+
+    return national_level_hemoglobin_data
+
+
+def get_pregnant_lactating_women_location_weights(key: str, location: str):
+    #  WRA * (ASFR + (ASFR * SBR) + incidence_c996 + incidence_c374)
```
Comment (rmudambi) [inline src/vivarium_gates_iv_iron/data/loader.py:485]: I know this is a question for Ali, but unless I'm misunderstanding what this quantity represents, I'd expect the numerator to be `WRA * incidence_pregnancy * duration_pregnancy`, while this is just `WRA  * incidence_pregnancy`.

File: src/vivarium_gates_iv_iron/components/hemoglobin.py [hunk 4]
```diff
@@ -16,6 +21,70 @@ def __init__(self):
...
+            loc_mean = mean.copy()
+            loc_stddev = stddev.copy()
+            loc_mean["country"] = country
+            loc_stddev["country"] = country
+            mean_dfs.append(loc_mean)
+            stddev_dfs.append(loc_stddev)
+            location_weights[country] = 1/3  # will be pop of country over pop of region
```
Comment (rmudambi) [inline src/vivarium_gates_iv_iron/components/hemoglobin.py:42]: If this is a stand-in value, add a todo comment so this is easy to find and doesn't get lost.

File: src/vivarium_gates_iv_iron/components/hemoglobin.py [hunk 5]
```diff
@@ -16,6 +21,70 @@ def __init__(self):
...
+        index_columns = ["country", "sex", "age_start", 'age_end', 'year_start', 'year_end']
+        mean = pd.concat(mean_dfs).set_index(index_columns)["value"].rename("mean")
+        stddev = pd.concat(stddev_dfs).set_index(index_columns)["value"].rename("stddev")
+        # <<<--- TODO replace
+        distribution_parameters = pd.concat([mean, stddev], axis=1).reset_index()
+        # TODO: look in Risk code to make sure we have canonical exposure naming
```
Comment (rmudambi) [inline src/vivarium_gates_iv_iron/components/hemoglobin.py]: You should address these canonical naming todos before merging.

**Review by rmudambi — CHANGES_REQUESTED** [review]

File: src/vivarium_gates_iv_iron/components/hemoglobin.py (hunk 2 above)
Comment (mattkappel) [reply src/vivarium_gates_iv_iron/components/hemoglobin.py:35]: This is temporary code.

File: src/vivarium_gates_iv_iron/components/hemoglobin.py (hunk 4 above)
Comment (mattkappel) [reply src/vivarium_gates_iv_iron/components/hemoglobin.py:42]: It's in a block that has a TODO to replace the code. Sorry for not making that clear.

File: src/vivarium_gates_iv_iron/components/hemoglobin.py (hunk 5 above)
Comment (mattkappel) [reply src/vivarium_gates_iv_iron/components/hemoglobin.py]: Fair point!

File: src/vivarium_gates_iv_iron/data/loader.py (hunk 3 above)
Comment (mattkappel) [reply src/vivarium_gates_iv_iron/data/loader.py:485]: This is the calculation of the population for a location, which we divide by the larger regional population to arrive at the location weight. The [research reference](https://vivarium-research.readthedocs.io/en/latest/concept_models/vivarium_iv_iron/maternal_model/concept_model.html#hemoglobin-distribution-weighting-strategy) in the above description describes this.

File: src/vivarium_gates_iv_iron/components/hemoglobin.py (hunk 5 above)
Comment (mattkappel) [reply src/vivarium_gates_iv_iron/components/hemoglobin.py]: Corrected with commit.

**Review by rmudambi — APPROVED** [review]

**Pull request merged**